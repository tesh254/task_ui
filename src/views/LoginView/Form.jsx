import React from "react";

const LoginForm = ({ handleSubmit, onChange, loading }) => (
  <form onSubmit={handleSubmit}>
    <h1 className="title is-3">Log In</h1>
    <span>
      Use phonenumber: 0705181707
          password: test254
    </span>
    <div className="field">
      <p className="control">
        <input
          className="input"
          name="phonenumber"
          type="text"
          placeholder="Phone Number"
          onChange={onChange}
        />
      </p>
    </div>
    <div className="field">
      <p className="control">
        <input
          className="input"
          type="password"
          name="password"
          placeholder="Password"
          onChange={onChange}
        />
      </p>
    </div>
    <div className="field">
      <p className="control">
        <button type="submit" className="button is-success is-fullwidth">
          Login
        </button>
      </p>
    </div>
  </form>
);

export default LoginForm;
