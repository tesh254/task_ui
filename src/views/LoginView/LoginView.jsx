import React from "react";
import LoginContainer from "../../containers/Login";

export default () => <LoginContainer />