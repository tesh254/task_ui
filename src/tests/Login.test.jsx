import React from "react";
import enzyme, { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import adapter from "enzyme-adapter-react-16";
import sinon from "sinon";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import LoginForm from "../views/LoginView/Form";
import LoginContainer from "../containers/Login";
import LoginView from "../views/LoginView";
import { Provider } from "react-redux";
import loginUser from "../redux/actions/auth/login";

enzyme.configure({ adapter: new adapter() });

const mockStore = configureStore([]);

function shallowSetup() {
  const props = {
    handleSubmit: jest.fn(),
    onChange: jest.fn(),
    loading: false
  };

  const enzymeWrapper = shallow(<LoginForm {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe("Shallow to render login form", () => {
  it("should render a login form", () => {
    const { enzymeWrapper } = shallowSetup();

    expect(enzymeWrapper.find("h1").hasClass("title is-3")).toBe(true);
    expect(
      enzymeWrapper.find("button").hasClass("button is-success is-fullwidth")
    ).toBe(true);
  });

  it("renders correctly", () => {
    const { enzymeWrapper } = shallowSetup();

    expect(toJSON(enzymeWrapper)).toMatchSnapshot();
  });
});

describe("Test Login Container", () => {
  let store;
  let component;
  let viewcomponent

  beforeEach(() => {
    store = mockStore({});

    const props = {
      loginUser: jest.fn(),
      isLoading: false
    };

    // sinon.spy(LoginContainer.prototype, "onSubmit")

    component = mount(
      <Provider store={store}>
        <MemoryRouter>
          <LoginContainer {...props} />
        </MemoryRouter>
      </Provider>
    );

    viewcomponent = mount(
      <Provider store={store}>
        <MemoryRouter>
          <LoginView {...props} />
        </MemoryRouter>
      </Provider>
    )
  });
  it("should render login container correctly", () => {
    expect(toJSON(component)).toMatchSnapshot();
  });

  it("should render login view correctly", () => {
    expect(toJSON(viewcomponent)).toMatchSnapshot();
  });
});
