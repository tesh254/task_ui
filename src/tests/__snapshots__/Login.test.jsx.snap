// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow to render login form renders correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h1
    className="title is-3"
  >
    Log In
  </h1>
  <div
    className="field"
  >
    <p
      className="control"
    >
      <input
        className="input"
        name="phonenumber"
        onChange={[MockFunction]}
        placeholder="Phone Number"
        type="text"
      />
    </p>
  </div>
  <div
    className="field"
  >
    <p
      className="control"
    >
      <input
        className="input"
        name="password"
        onChange={[MockFunction]}
        placeholder="Password"
        type="password"
      />
    </p>
  </div>
  <div
    className="field"
  >
    <p
      className="control"
    >
      <button
        className="button is-success is-fullwidth"
        type="submit"
      >
        Login
      </button>
    </p>
  </div>
</form>
`;

exports[`Test Login Container should render login container correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "4ldw89",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "4ldw89",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(LoginContainer))
        isLoading={false}
        loginUser={[MockFunction]}
      >
        <Connect(LoginContainer)
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "4ldw89",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "4ldw89",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isLoading={false}
          location={
            Object {
              "hash": "",
              "key": "4ldw89",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          loginUser={[MockFunction]}
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <LoginContainer
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "4ldw89",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "4ldw89",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isLoading={true}
            location={
              Object {
                "hash": "",
                "key": "4ldw89",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            loginUser={[Function]}
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <section
              className="has-text-centered"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <LoginForm
                handleSubmit={[Function]}
                loading={true}
                onChange={[Function]}
              >
                <form
                  onSubmit={[Function]}
                >
                  <h1
                    className="title is-3"
                  >
                    Log In
                  </h1>
                  <div
                    className="field"
                  >
                    <p
                      className="control"
                    >
                      <input
                        className="input"
                        name="phonenumber"
                        onChange={[Function]}
                        placeholder="Phone Number"
                        type="text"
                      />
                    </p>
                  </div>
                  <div
                    className="field"
                  >
                    <p
                      className="control"
                    >
                      <input
                        className="input"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                      />
                    </p>
                  </div>
                  <div
                    className="field"
                  >
                    <p
                      className="control"
                    >
                      <button
                        className="button is-success is-fullwidth"
                        type="submit"
                      >
                        Login
                      </button>
                    </p>
                  </div>
                </form>
              </LoginForm>
            </section>
          </LoginContainer>
        </Connect(LoginContainer)>
      </withRouter(Connect(LoginContainer))>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`Test Login Container should render login view correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "66z1mc",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "66z1mc",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <_default
        isLoading={false}
        loginUser={[MockFunction]}
      >
        <withRouter(Connect(LoginContainer))>
          <Connect(LoginContainer)
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "66z1mc",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "66z1mc",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "66z1mc",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <LoginContainer
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "key": "66z1mc",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "66z1mc",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              isLoading={true}
              location={
                Object {
                  "hash": "",
                  "key": "66z1mc",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              loginUser={[Function]}
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <section
                className="has-text-centered"
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <LoginForm
                  handleSubmit={[Function]}
                  loading={true}
                  onChange={[Function]}
                >
                  <form
                    onSubmit={[Function]}
                  >
                    <h1
                      className="title is-3"
                    >
                      Log In
                    </h1>
                    <div
                      className="field"
                    >
                      <p
                        className="control"
                      >
                        <input
                          className="input"
                          name="phonenumber"
                          onChange={[Function]}
                          placeholder="Phone Number"
                          type="text"
                        />
                      </p>
                    </div>
                    <div
                      className="field"
                    >
                      <p
                        className="control"
                      >
                        <input
                          className="input"
                          name="password"
                          onChange={[Function]}
                          placeholder="Password"
                          type="password"
                        />
                      </p>
                    </div>
                    <div
                      className="field"
                    >
                      <p
                        className="control"
                      >
                        <button
                          className="button is-success is-fullwidth"
                          type="submit"
                        >
                          Login
                        </button>
                      </p>
                    </div>
                  </form>
                </LoginForm>
              </section>
            </LoginContainer>
          </Connect(LoginContainer)>
        </withRouter(Connect(LoginContainer))>
      </_default>
    </Router>
  </MemoryRouter>
</Provider>
`;
